components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +235,90 @@ func NewAdxl345(
 			}
 		}
 	})
-
+	sensor.interruptsFound = make(map[string]int)
+	sensor.readInterrupts(sensor.cancelContext)
+	err = sensor.configureInterruptRegisters(ctx)
+	if err != nil {
+		sensor.cancelFunc()
+		return nil, err
+	}
+	ticksChan := make(chan board.Tick)
+	if sensor.echoInterrupt1 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt1, ticksChan)
+	}
+	if sensor.echoInterrupt2 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt2, ticksChan)
+	}
 	return sensor, nil
 }
 
+func (adxl *adxl345) startInterruptPolling(interrupt board.DigitalInterrupt, ticksChan chan board.Tick) {
+	utils.PanicCapturingGo(func() {
+		interrupt.AddCallback(ticksChan)
+		defer interrupt.RemoveCallback(ticksChan)
+
+		for {
+			select {
+			case <-adxl.cancelContext.Done():
+				return
+			case tick := <-ticksChan:
+				if tick.High {
+					func() {
+						adxl.readInterrupts(adxl.cancelContext)
+					}()
+				}
+			}
+		}
+	})
+}
+
+func getInterruptConfigurations(cfg *AttrConfig) map[byte]byte {
+	var intEnabled byte
+	var intMap byte
+
+	if cfg.FreeFall != nil {
+		intEnabled += interruptBitPosition[FreeFall]
+		intMap += interruptBitPosition[FreeFall] & byte(cfg.FreeFall.AccelerometerPin<<2)
```

Comment:
I had thought that `AccelerometerPin` will be either 1 or 2, depending on which pin you've wired up. If that's right, we're bitwise-anding things with either 0b100 or 0b1000, and it feels very strange to me to change the bitmask like that. I suspect what you're really doing is setting the bit to 1 for pin INT1 and 0 for pin INT2, but the approach you've taken doesn't make that obvious. How about this instead?

```go
if cfg.FreeFall.AccelerometerPin == 1 {
    intMap |= interruptBitPosition[FreeFall]
} else {
    // Clear the freefall bit in the map to send the signal to pin INT2.
    intMap &^= interruptBitPosition[FreeFall]
}
```
(I did not know about the `&^=` operator before I started writing this comment. It's possible that's too esoteric and should be avoided for readability reasons.)